### ISSUE TITLE
Find a way for users to get to a selected carpark

### ISSUE THREAD

**Issue by Nelsontwb93** [Tzyinc/sg-carpark-bot#3]
Lot capacity
- demand and availability 

Geolocation or postal code

**Comment by Tzyinc** [comment]
these data exist, will find a way to integrate smoothly! lot capacity has already been mentioned in my other issue, so I'm renaming your title regarding the geolocation and postal code. I think it's not so much so you want the geoloc/postalcode as to how to get there right? If not, do elaborate why you'd want the geoloc/postalcode 

my priority is to reduce the wall of text so that people can actually use it while driving

**Comment by Nelsontwb93** [comment]
1 is to find a way to the decided carpark. 
another: lets say i would like to plan a trip to a hawker centre (or any other place). I would want to balance between which car park will have slot and cheapest vs the distance it needs to walk to the hawker centre.

**Comment by Tzyinc** [comment]
From my understanding I think you want to calculate a heuristic based on distance, cost and lots availability, and then make a selection based on this heuristic. 

Unfortunately my goal for this bot is to be as lightweight as possible, so as maximise number of users without using a huge server. With your proposed enhancement, I think it would be too computationally expensive for each request and I will probably never add such a feature. As such, I will be closing this issue.

You are however welcome to clone the project and make a smarter better bot if you want to develop it yourself! :)  

A few issues I foresee you might face if you want to implement that feature:
- The dataset provided by LTA (CarParkRates.csv) is in an absolute mess, the rates are in different formats and have arbitrary exceptions such that it's not trivial to write a crawler to clean the rates and put them in nice numerical values
- dataset also does not include lat and long, which I have worked around by pre-processing via google maps to geocode the carparks based on names. This means that my latlong do not point to the carparks, but the establishments itself (e.g. Vivocity latlong as opposed to the Vivocity carpark lat long)
- lot availability api data set does not match the carpark rates api in any way, be it having different names, places like Vivocity having 2 carparks in the lot avail but only 1 in the rates api, or the lot avail api just simply only has data on 30 carparks

on the other hand, to solve some of these problems I'm facing with my own implementation, I have just committed an update where my pre-processed data on the lots is now in a csv file as opposed to a JSON object, so that non-programmers can add and modify the data, this way I can get my non-programmer friends to clean up my data csv, add new carparks and rates, as well as enter the lot availability ID manually such that it acts as a "foreign key" to the lots availability api. Not an automated solution, but this was the easiest and fastest way I could find to fix these issues
